sensor/kubernetes/listener/resources/dispatcher.go

```diff
@@ -65,16 +65,17 @@ func NewDispatcherRegistry(
 	namespaces *orchestratornamespaces.OrchestratorNamespaces,
 	credentialsManager awscredentials.RegistryCredentialsManager,
 	traceWriter io.Writer,
+	storeProvider *InMemoryStoreProvider,
```

Comment:
This should be maybe an interface? `InMemoryStoreProvider` is an implementation.
No need to do it now, but I think this is what we want to have long term.

Reply:
The downside of making this an interface is that it prevents us for having the _private_ and _public_ split of stores. This only works because the provider exposes the public interface of stores using methods, but allows the listeners component to access the concrete store implementation, and hence their private methods.